### PR TITLE
Downloadable acls on Cisco

### PR DESCRIPTION
**Pull request by julsemaan**
Adds the downloadable acls feature on the cisco catalyst 2960.
Also adds it in the admin interface (that could be reworked)

This can then be used to push the web auth acls using packetfence instead of configuring them in the equipment.

For review and then merge if the admin interface is ok.

### CODE REVIEW

**Comment by extrafu** [comment]
Please provide a couple of examples (2-3) in our 2960 documentation.

File: lib/pf/Switch/Cisco/Catalyst_2960.pm [hunk 1]
```diff
@@ -499,6 +501,63 @@ sub wiredeauthTechniques {
...
+            'Tunnel-Medium-Type' => $RADIUS::ETHERNET,
+            'Tunnel-Type' => $RADIUS::VLAN,
+            'Tunnel-Private-Group-ID' => $vlan,
+        };
+    }
+
+    
+    if ( isenabled($self->{_AccessListMap}) && $self->supportsAccessListBasedEnforcement ){
+        use Data::Dumper;
+        $logger->info(Dumper($self->{_access_lists}));
```
Comment (fdurand) [inline lib/pf/Switch/Cisco/Catalyst_2960.pm]: To remove

File: lib/pf/Switch/Cisco/Catalyst_2960.pm [hunk 2]
```diff
@@ -499,6 +501,63 @@ sub wiredeauthTechniques {
...
+        if( defined($user_role) && $user_role ne ""){
+            my $access_list = $self->getAccessListByName($user_role);
+            my @av_pairs;
+            while($access_list =~ /([^\n]+)\n?/g){
+                push(@av_pairs, "ip:inacl#101=$1");
+            } 
+            $logger->info(Dumper(\@av_pairs));
```
Comment (fdurand) [inline lib/pf/Switch/Cisco/Catalyst_2960.pm]: Same

File: lib/pf/Switch/Cisco/Catalyst_2960.pm [hunk 3]
```diff
@@ -499,6 +501,63 @@ sub wiredeauthTechniques {
...
+
+    
+    if ( isenabled($self->{_AccessListMap}) && $self->supportsAccessListBasedEnforcement ){
+        use Data::Dumper;
+        $logger->info(Dumper($self->{_access_lists}));
+        if( defined($user_role) && $user_role ne ""){
+            my $access_list = $self->getAccessListByName($user_role);
+            my @av_pairs;
+            while($access_list =~ /([^\n]+)\n?/g){
+                push(@av_pairs, "ip:inacl#101=$1");
```
Comment (fdurand) [inline lib/pf/Switch/Cisco/Catalyst_2960.pm]: Better to use something like $self->returnAccessListAttribute()

File: lib/pf/Switch/Cisco/Catalyst_2960.pm [hunk 4]
```diff
@@ -499,6 +501,63 @@ sub wiredeauthTechniques {
...
+            } 
+            $logger->info(Dumper(\@av_pairs));
+            $radius_reply_ref->{'Cisco-AVPair'} = \@av_pairs; 
+        }
```
Comment (fdurand) [inline lib/pf/Switch/Cisco/Catalyst_2960.pm:538]: $logger->info("blabla");

**Comment by julsemaan** [comment]
Code cleanup + documentation made.

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 5]
```diff
@@ -1021,6 +1021,39 @@ PacketFence switch configuration
 * Set your registered roles to 'registered'
 * Screenshots of this configuration are available in the Cisco WLC section of this guide.
 
+Downloadable ACLs
++++++++++++++++++
+
+The Catalyst 2960 supports RADIUS pushed ACLs which means that you can define the ACLs centrally in PacketFence without configuring them in your switches and their rules will be applied to the switch during the authentication.
+
+These ACLs are defined per node category like the VLANs which means you can define different ACLs for your registration VLAN, production VLAN, guest VLAN, etc.
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: defined by role

**Comment by julsemaan** [comment]
Review + testing is done
Merge this for 4.4 ?

File: NEWS.asciidoc [hunk 6]
```diff
@@ -24,6 +24,7 @@ New Features
 * Added support for unreg date with dynamic year
 * Added support for NetGear FSM7328S switches
 * Added new network profile filter
+* Added support for Dynamic Access lists on the Cisco Catalyst 2960
```
Comment (fdurand) [inline NEWS.asciidoc:29]: Cisco Catalyst 2960 2970 3550 3560 3750 4500 6500

**Pull request merged**